beacon-chain/state/v1/state_trie.go

```diff
@@ -79,6 +118,48 @@ func InitializeFromProtoUnsafe(st *ethpb.BeaconState) (*BeaconState, error) {
 	return b, nil
 }
 
+func Initialize(vals []*ethpb.Validator) (*BeaconState, error) {
```

Comment:
Function added to allow initializing without the proto state

Reply:
why only with validators ?

Reply:
They are used to initialize the validator map, but this is also done in `SetValidators`.